src/stats/probs/cdf.rs

```diff
@@ -5,6 +5,106 @@
 
 //! Support for discrete probability distributions in terms of cumulative distribution
 //! functions (CDF).
+//!
+//! # Examples
+//!
+//! Example usage of all CDF functions:
+//! ```
+//! use std::ops::Range;
+//! use ordered_float::NotNan;
+//! use bio::stats::probs::{LogProb, Prob};
+//! use bio::stats::probs::cdf::{CDF, Entry};
+//! use approx::assert_relative_eq;
+//! // pmf1 is an example PMF with `LogProb(0.0)` at `0`, with `LogProb(0.1)`
+//! // at `{1, 2, ..., 8}` and LogProb(0.2) at `10`
+//! let mut pmf1 = vec![
+//!     Entry::new( 0, LogProb( (0.0 as f64).ln() ) )
+//! ];
+//! for i in 1..=8 {
+//!     pmf1.push(
+//!         Entry::new( i, LogProb( (0.1 as f64).ln() ) )
+//!     );
+//! }
+//! pmf1.push( Entry::new( 10, LogProb( (0.2 as f64).ln() ) ) );
+//!
+//! // create the cumulative distribution function from the probability mass function
+//! let cdf = CDF::from_pmf(pmf1.clone());
+//! assert_relative_eq!(
+//!     *cdf.get(&0).unwrap(),
+//!     (0.0 as f64).ln(),
+//!     epsilon = 0.0
+//! );
+//! assert_relative_eq!(
+//!     *cdf.get(&3).unwrap(),
+//!     (0.3 as f64).ln(),
+//!     epsilon = 0.0000000001
+//! );
+//!
+//! // get back the original probability mass value at 7
+//! assert_relative_eq!(
+//!     *cdf.get_pmf(&7).unwrap(),
+//!     (0.1 as f64).ln(),
+//!     epsilon = 0.00001
+//! );
+//!
+//! // Check that cdf sums up to 1.0
+//! assert_relative_eq!(
+//!     f64::from(cdf.total_prob()),
+//!     (1.0 as f64).ln(),
+//!     epsilon = 0.0
+//! );
+//!
+//! // copy a CDF via its iter() function
+//! let mut cdf_copy = CDF::from_cdf(cdf.iter().cloned());
+//! assert_eq!(cdf.len(), cdf_copy.len());
+//!
+//! // get the maximum a posteriori probability estimate
+//! assert_eq!( cdf_copy.map().unwrap(), &10 );
```

Comment:
Hm, `map` is quite unfortunate choice for a function name that doesn't map but return the maximum a posteriori estimate... :D

Reply:
Hey, I'm just providing examples here. I absolve myself from any responsibilty for function naming that this exposes... :D